modules/spring-config/src/test/java/org/mule/runtime/config/internal/el/DataWeaveExtendedExpressionLanguageAdaptorFactoryBeanTestCase.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.config.internal.el;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.withSettings;
+
+import org.hamcrest.core.IsInstanceOf;
+import org.mule.runtime.api.lifecycle.Disposable;
+import org.mule.runtime.core.internal.el.ExtendedExpressionLanguageAdaptor;
+
+import io.qameta.allure.Issue;
+import org.junit.Test;
+import org.springframework.beans.factory.DisposableBean;
+
+@Issue("W-10672687")
+public class DataWeaveExtendedExpressionLanguageAdaptorFactoryBeanTestCase {
+
+  @Test
+  public void whenDataWeaveExtendedExpressionLanguageAdaptorFactoryBeanIsDistroyedTheAdapatorIsDisposed() throws Exception {
+    TestDataWeaveExtendedExpressionLanguageAdaptorFactoryBean factoryBean =
```

Comment:
this is testing the implementation rather than the actual behavior. Can't we instead hace a test that disposes a MuleContext and verifies that the adaptor has been disposed?

Reply:
It tests that when the destroy method of the factory is invoked it also disposes the singleton bean created which is the way to do achieve the disposal of the singleton according to spring java docs.
I'll try anyway to create an integration test.